src/period-calculation/getYearlyPeriods.ts

```diff
@@ -0,0 +1,167 @@
+import { Temporal } from "@js-temporal/polyfill";
+import { SupportedCalendar } from "../types";
+import { formatYyyyMmDD, isCustomCalendar } from "../utils/helpers";
+import localisationHelpers from "../utils/localisationHelpers";
+import {
+  FixedPeriod,
+  GeneratedPeriodsFunc,
+  PeriodIdentifier,
+} from "./fixed-periods";
+import { monthNumbers } from "./getMonthlyPeriods";
+
+export const getYearlyPeriods: GeneratedPeriodsFunc = ({
+  year,
+  calendar,
+  periodType,
+  yearsCount = 10,
+  locale,
+}) => {
+  const month = getMonth(periodType);
+  const currentYear = Temporal.PlainDate.from({
+    year,
+    month,
+    day: calendar.toString() === "nepali" ? 14 : 1,
+    calendar,
+  });
+
+  const years: FixedPeriod[] = [];
+
+  for (let i = 0; i < yearsCount; i++) {
+    const dateToAdd = currentYear.subtract({ years: i });
+    const value = buildValue({
+      periodType,
+      year: dateToAdd.year,
+      month,
+    });
+    const year: FixedPeriod = {
+      id: value,
+      iso: value,
+      name: buildLabel(periodType, dateToAdd, {
+        locale: locale || "en",
+        calendar,
+      }),
+      ...buildStartAndEndDates(dateToAdd),
+    };
+    if (isFinancialYear(periodType)) {
+      delete year.iso;
+    }
+    years.push(year);
+  }
+  return years;
+};
+
+const buildStartAndEndDates = (date: Temporal.PlainDate) => {
+  const endDate = Temporal.PlainDate.from({
+    year: date.year + 1,
+    month: date.month,
+    day: 1,
+    calendar: date.calendar,
+  }).subtract({ days: 1 });
+  return {
+    startDate: formatYyyyMmDD(date, "startOfMonth"),
+    endDate: formatYyyyMmDD(endDate, "endOfMonth"),
+  };
+};
+
+const getMonth = (periodType: string) => {
+  const yearType = periodType?.replace("FY", "").toUpperCase();
```

Comment:
I'm not sure if I get the optional chaining here? There are a couple of places in this file where I would assume the optional chaining would be unnecessary (e.g. also below with `monthInfo?.value || value` (e.g. I'm not sure how periodType or monthInfo would be undefined)